### ISSUE TITLE
affiliates and classroom 0.2.0: Ambiguous occurrence ‘Link’

### ISSUE THREAD

**Issue by juhp** [brendanhay/gogol#61]
In Stackage Nightly:

```
Preprocessing library gogol-affiliates-0.2.0...
[ 1 of 14] Compiling Network.Google.Affiliates.Types.Sum ( gen/Network/Google/Affiliates/Types/Sum.hs, dist/build/Network/Google/Affiliates/Types/Sum.o )
[ 2 of 14] Compiling Network.Google.Affiliates.Types.Product ( gen/Network/Google/Affiliates/Types/Product.hs, dist/build/Network/Google/Affiliates/Types/Product.o )

gen/Network/Google/Affiliates/Types/Product.hs:380:8: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Affiliates/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Affiliates.Types.Product.Link’,
                             defined at gen/Network/Google/Affiliates/Types/Product.hs:310:1

gen/Network/Google/Affiliates/Types/Product.hs:407:22: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Affiliates/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Affiliates.Types.Product.Link’,
                             defined at gen/Network/Google/Affiliates/Types/Product.hs:310:1

gen/Network/Google/Affiliates/Types/Product.hs:412:26: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Affiliates/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Affiliates.Types.Product.Link’,
                             defined at gen/Network/Google/Affiliates/Types/Product.hs:310:1

gen/Network/Google/Affiliates/Types/Product.hs:418:28: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Affiliates/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Affiliates.Types.Product.Link’,
                             defined at gen/Network/Google/Affiliates/Types/Product.hs:310:1
:
```

```
Preprocessing library gogol-classroom-0.2.0...
[ 1 of 46] Compiling Network.Google.Classroom.Types.Sum ( gen/Network/Google/Classroom/Types/Sum.hs, dist/build/Network/Google/Classroom/Types/Sum.o )
[ 2 of 46] Compiling Network.Google.Classroom.Types.Product ( gen/Network/Google/Classroom/Types/Product.hs, dist/build/Network/Google/Classroom/Types/Product.o )

gen/Network/Google/Classroom/Types/Product.hs:641:8: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Classroom/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Classroom.Types.Product.Link’,
                             defined at gen/Network/Google/Classroom/Types/Product.hs:625:1

gen/Network/Google/Classroom/Types/Product.hs:650:24: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Classroom/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Classroom.Types.Product.Link’,
                             defined at gen/Network/Google/Classroom/Types/Product.hs:625:1

gen/Network/Google/Classroom/Types/Product.hs:657:15: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Classroom/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Classroom.Types.Product.Link’,
                             defined at gen/Network/Google/Classroom/Types/Product.hs:625:1

gen/Network/Google/Classroom/Types/Product.hs:661:17: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Classroom/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Classroom.Types.Product.Link’,
                             defined at gen/Network/Google/Classroom/Types/Product.hs:625:1

gen/Network/Google/Classroom/Types/Product.hs:664:19: error:
    Ambiguous occurrence ‘Link’
    It could refer to either ‘Network.Google.Prelude.Link’,
                             imported from ‘Network.Google.Prelude’ at gen/Network/Google/Classroom/Types/Product.hs:21:1-39
                             (and originally defined in ‘servant-0.10:Servant.Utils.Links’)
                          or ‘Network.Google.Classroom.Types.Product.Link’,
                             defined at gen/Network/Google/Classroom/Types/Product.hs:625:1
:
:
```

**Comment by brendanhay** [comment]
Should be resolved by [gogol-core-0.2.0.1](http://hackage.haskell.org/package/gogol-core) on Hackage.

**Comment by juhp** [comment]
FYI @snoyberg
